src/cljs/rems/application.cljs

```diff
@@ -438,6 +439,142 @@
+   (update db ::selected-3rd-party-reviewers disj reviewer)))
+
+(rf/reg-sub
+ ::selected-3rd-party-reviewers
+ (fn [db _]
+   (::selected-3rd-party-reviewers db)))
+
+(rf/reg-fx
+ ::fetch-potential-3rd-party-reviewers
+ (fn [[user]]
+   (fetch-potential-3rd-party-reviewers user)))
+
+(rf/reg-event-db
+ ::set-potential-3rd-party-reviewers
+ (fn [db [_ reviewers]]
+   (assoc db ::potential-3rd-party-reviewers (for [reviewer reviewers]
+                                               (assoc reviewer :display (str (:name reviewer) " (" (:email reviewer)")"))))))
+
+(rf/reg-event-fx
+ ::start-fetch-potential-3rd-party-reviewers
+ (fn [{:keys [db]} [_]]
+   {::fetch-potential-3rd-party-reviewers [(get-in db [:identity :user])]}))
+
+(rf/reg-sub
+ ::potential-3rd-party-reviewers
+ (fn [db _]
+   (::potential-3rd-party-reviewers db)))
+
+(rf/reg-event-db
+ ::set-review-comment
+ (fn [db [_ value]]
+   (assoc db ::review-comment value)))
+
+(rf/reg-sub
+ ::review-comment
+ (fn [db _]
+   (::review-comment db)))
+
+(defn- send-3rd-party-review-request [reviewers user application-id round comment]
+  (PUT "/api/application/review_request"
+       {:format :json
+        :params {:application-id application-id
+                 :round round
+                 :comment comment
+                 :recipients (map :userid reviewers)}
+        :handler (fn [resp]
+                   (rf/dispatch [::send-3rd-party-review-request-success true]))}))
+
+(rf/reg-event-fx
+ ::send-3rd-party-review-request
+ (fn [{:keys [db]} [_ reviewers comment]]
+   (let [application-id (get-in db [:application :application :id])
+         round (get-in db [:application :application :curround])
+         user (get-in db [:identity :user])]
+     (rf/dispatch [::set-review-comment ""])
+     (send-3rd-party-review-request reviewers user application-id round comment)
+     {})))
+
+(rf/reg-event-db
+ ::send-3rd-party-review-request-success
+ (fn [db [_ value]]
+   (assoc db ::send-3rd-party-review-request-message value)))
+
+(rf/reg-sub
+ ::send-3rd-party-review-request-message
+ (fn [db _]
+   (::send-3rd-party-review-request-message db)))
+
+(defn scroll-to-top! []
+  (.setTimeout js/window #(.scrollTo js/window 0 0) 500)) ;; wait until faded out
+
+(defn- review-request-modal []
+  (let [selected-3rd-party-reviewers @(rf/subscribe [::selected-3rd-party-reviewers])
+        potential-3rd-party-reviewers @(rf/subscribe [::potential-3rd-party-reviewers])
+        review-comment @(rf/subscribe [::review-comment])]
+    [:div.modal.fade {:id "review-request-modal" :role "dialog" :aria-labelledby "confirmModalLabel" :aria-hidden "true"}
+     [:div.modal-dialog {:role "document"}
+      [:div.modal-content
+       [:div
+        [:div.modal-header
+         [:h5#confirmModalLabel.modal-title (text :t.applications/request-review)]
+         [:button.close {:type "button" :data-dismiss "modal" :aria-label (text :t.actions/cancel)}
+          [:span {:aria-hidden "true"} "\u00D7"]]]
```

Comment:
[minor] should this be an atom?

Reply:
Lets do that if/when we get another modal.